### PR TITLE
update docs with examples to set valid platform family

### PR DESCRIPTION
**Pull request by sathish-progress**
Updates the missing documentation for OS platform family 

## Description

- Provides clear example of configuring OS families that the transport can support

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New content (non-breaking change)
- [ ] Breaking change (a content change which would break existing functionality or processes)

## Checklist:

- [X] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: docs/plugins.md [hunk 1]
```diff
@@ -144,3 +146,12 @@ def platform
  )
 end
 ```
+Example config for transport that supports `unix` and `windows` OS,
+
+```Ruby
+def platform
+  Train::Platforms.name("local-rot13").in_family("unix")
```
Comment (Vasu1105) [inline docs/plugins.md:153]: I did not understood what "local-rot13" here is. Also what is `Example config`.

Comment (sathish-progress) [reply docs/plugins.md:153]: This is the recommended sample train plugin https://github.com/inspec/train/blob/main/examples/plugins/train-local-rot13/lib/train-local-rot13/platform.rb

Comment (sathish-progress) [reply docs/plugins.md:153]: `Example config` == `Configuration Example`

File: docs/plugins.md [hunk 2]
```diff
@@ -113,6 +113,8 @@ If your transport is OS based and has the option to read a file you can set this
 If your transport is OS based and has the option to run a command you can set this method. It is expected to return a `CommandResult` class here to be used upstream in InSpec. Currently the command resource is restricted to Unix and Windows platforms. Caching is enabled by default for this method.  
 You can optionally receive an options hash as well as the command string. This is intended for options that apply only to the current instance of running a command. For example, applying a timeout to this command execution.
 
+**In order to run file and commands OS based, please make sure the platform family belongs to a valid OS. Example config is provided below.
```
Comment (Vasu1105) [inline docs/plugins.md:116]: The sentence needs to be reconstructed a bit it seems.

File: docs/plugins.md [hunk 3]
```diff
@@ -113,6 +113,8 @@ If your transport is OS based and has the option to read a file you can set this
 If your transport is OS based and has the option to run a command you can set this method. It is expected to return a `CommandResult` class here to be used upstream in InSpec. Currently the command resource is restricted to Unix and Windows platforms. Caching is enabled by default for this method.  
 You can optionally receive an options hash as well as the command string. This is intended for options that apply only to the current instance of running a command. For example, applying a timeout to this command execution.
 
+**If your transport is OS based, do not force the platform family using `force_platform!`. The platform family will be detected dynamically based on the OS.
```
Comment (Vasu1105) [inline docs/plugins.md:116]: According to me this is a wrong place to have this note here. It should go under platform detection section and as per me platform detection area explains in first few lines.

Comment (sathish-progress) [reply docs/plugins.md:116]: okay I will remove this in that case..

**Review by ahasunos — APPROVED** [review]

**Review by Vasu1105 — APPROVED** [review]

**Review by Schwad — APPROVED** [review]

**Pull request merged**